frontend/PreferencesPanel.js

```diff
@@ -118,6 +122,26 @@ class PreferencesPanel extends React.Component<Props, State> {
               <EditIcon />
             </EditButton>
           </div>
+          <div>
+            <label>
+              <input
+                type="checkbox"
+                checked={hideSymbol}
+                onChange={this._changeHideSymbol}
+              />
+              Hide components with truthy <code>Symbol.for('react.devtools.hide')</code> property
+            </label>
+          </div>
+          <div>
+            <label>
+              <input
+                type="checkbox"
+                checked={hideDisplayNamed}
+                onChange={this._changeHideDisplayNamed}
+              />
+              Hide components with custom <code>displayName</code> property
```

Comment:
I don't understand why you'd want this particular functionality.

Reply:
Many HOC's creates wrapper components with custom displayName like `SomeWrapper(OriginalComponent)`
In contrast, regular components usually do not use dispalyName

I thought that such a simple rule is suitable for a draft, and better than nothing
This is the minimum that I would like to get in order to hide noisy HOC's
I can add/replace with other simple HOC recognition, if they work better.

Reply:
> Many HOC's creates wrapper components with custom displayName
>In contrast, regular components usually do not use dispalyName

There are many that are kind of like "presentation HOC" and assign displayName through babel plugin, i.e styled-components, emotion, jss, etc:
```jsx
const MyButton = styled.button`
   color: tomato;
`
// MyButton will have displayName: "MyButton" crafted by babel plugin
```
Although they are technically HOC i'd not want to hide them.

Reply:
I think the `displayName` attribute is too likely to cause false positives, as @asvetliakov points out, due to popular plug-ins like [`babel-plugin-transform-react-display-name`](https://www.npmjs.com/package/babel-plugin-transform-react-display-name). It also feels too arbitrary to me.